docs/src/resources/two-side-column.md

```diff
@@ -0,0 +1,519 @@
+         </tr>
+         <tr>
+<td>
+
+```
+    module foo (
+                    input  a,
+                    output b
+                )
+                assign b = a;
+            endmodule
+```
+
+</td>
+    <td>
+
+```scala mdoc
+    class Foo extends Module {
+    val a = Input(Bool())
+    val b = Output(Bool())
+    b := a
+    }
+```
+
+</td>
+    <td>Text comes here</td>
+         </tr>
+    <tr>
+<td>
+Text comes here
+</td>
+<td>
+
+```scala mdoc:invisible
+import Chisel.Queue
+import chisel3._
+import chisel3.util.DecoupledIO
+```
+
+```scala mdoc:silent
+class PassthroughGenerator(width: Int) extends Module {
+val io = IO(new Bundle {
+val in = Input(UInt(width.W))
+val out = Output(UInt(width.W))
+})
+io.out := io.in
+}
+println(getVerilogString(new PassthroughGenerator(10)))
+println(getVerilogString(new PassthroughGenerator(20)))
+```
+</td>
+<td>
+
+```
+module PassthroughGenerator(
+  input        clock,
+  input        reset,
+  input  [9:0] io_in,
+  output [9:0] io_out
+);
+  assign io_out = io_in; // @[main.scala 13:10]
+endmodule
+
+module PassthroughGenerator(
+input         clock,
+input         reset,
+input  [19:0] io_in,
+output [19:0] io_out
+);
+assign io_out = io_in; // @[main.scala 13:10]
+endmodule
+```
+</td>
+         </tr>
+    </table>
+</body>
+</html>
+
+#Parameterizing a Module
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+            <td><b style="font-size:30px">Generated Verilog</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+module ParameterizedWidthAdder(
+
+input [in0Width-1:0] in0,
+input [in1Width-1:0] in1,
+output [sumWidth-1:0] sum
+);
+parameter in0Width = 8;
+parameter in1Width = 1;
+parameter sumWidth = 9;
+
+assign sum = in0 + in1;
+
+endmodule
+```
+
+</td>
+<td>
+
+```
+class ParameterizedWidthAdder(in0Width: Int, in1Width: Int, sumWidth: Int) extends Module {
+  require(in0Width >= 0)
+  require(in1Width >= 0)
+  require(sumWidth >= 0)
+  val io = IO(new Bundle {
+    val in0 = Input(UInt(in0Width.W))
+    val in1 = Input(UInt(in1Width.W))
+    val sum = Output(UInt(sumWidth.W))
+  })
+  // a +& b includes the carry, a + b does not
+  io.sum := io.in0 +& io.in1
+}
+```
+</td>
+             <td>Text comes here</td>
+</tr>
+<tr>
+<td>
+
+```
+module TestBench;
+wire [31:0] sum;
+ParameterizedWidthAdder  #(32, 32, 32) my32BitAdderWithTruncation (32'b0, 32'b0, sum);
+endmodule
+```
+</td>
+            <td>
+
+```
+val my32BitAdderWithTruncation = Module(new ParameterizedWidthAdder(32, 32, 32)
+```
+</td>
+             <td>Text comes here</td>
+         </tr>
+ <tr>
+<td>
+Text comes here
+</td>
+<td>
+
+```scala mdoc:silent
+class MyModule extends Module {
+  val io = IO(new Bundle {
+    val in  = Input(UInt(4.W))
+    val out = Output(UInt(4.W))
+  })
+
+val two  = 1 + 1
+println(two)
+val utwo = 1.U + 1.U
+println(utwo)
+
+io.out := io.in
```

Comment:
If we want to do something to show a lot of operators, we could do something like:

```
class OperatorExampleModule extends Module {

val a, b, c, d, e, f, g = Output(UInt(32.W))

a := x + y
b := x - y 
...
}
```

Reply:
scastie: https://scastie.scala-lang.org/BVlfJ5rySzehHip2WESOoA

Reply:
I keep getting “Illegal start of simple pattern” error. 
here is the scastie link: https://scastie.scala-lang.org/Shorla/3tMpKXSiQ762Rqt2k0Ag8A/2

Reply:
we figured out this was just an extra `,` after the `e`

Reply:
Hi, here is the scastie for what I have done on operators: https://scastie.scala-lang.org/Shorla/3tMpKXSiQ762Rqt2k0Ag8A/9

There is an error fixing Cat and Mux

Reply:
Also, I think we should show the bit select operators ( `x(n)`, `x(n, m)`) from the better cheat sheet:
<img width="341" alt="Screen Shot 2022-01-13 at 9 20 03 AM" src="https://user-images.githubusercontent.com/17858596/149378152-8fdd8f1e-9f21-464d-a5aa-1580ab9da5e5.png">

Reply:
We can also show the Fill operator (from above) and the bit-reduction methods here:

<img width="371" alt="Screen Shot 2022-01-13 at 9 23 11 AM" src="https://user-images.githubusercontent.com/17858596/149378506-f3a7b8a0-8964-4838-a2bd-17f7db12699e.png">

Reply:
The updated operators sasctie: https://scastie.scala-lang.org/Shorla/Helf7JLnQkS3AbAdWyEBKA/6